Kuzzle.Tests/KuzzleTest.cs

```diff
@@ -0,0 +1,183 @@
+using Xunit;
+using Newtonsoft.Json.Linq;
+using KuzzleSdk.Exceptions;
+using Kuzzle.Tests.Protocol;
+using KuzzleSdk.Protocol;
+using KuzzleSdk.API;
+using System;
+using System.Threading.Tasks;
+using Moq;
+
+namespace Kuzzle.Tests {
+  public class KuzzleTest {
+    private readonly KuzzleSdk.Kuzzle _kuzzle;
+    private readonly ProtocolMock _protocol;
+
+    public KuzzleTest() {
+      _protocol = new ProtocolMock();
+      _kuzzle = new KuzzleSdk.Kuzzle(_protocol.MockedObject);
+    }
+
+    [Fact]
+    public void DispatchTokenExpiredTest() {
+      bool eventDispatched = false;
+
+      _kuzzle.TokenExpired += delegate() {
+        eventDispatched = true;
+      };
+
+      _kuzzle.DispatchTokenExpired();
+
+      Assert.Null(_kuzzle.AuthenticationToken);
```

Comment:
The token is null by default, so this fails to verify that dispatching the TokenExpired event unsets the token.